lib/workers/repository/process/lookup/index.ts

```diff
@@ -495,3 +472,85 @@ export async function lookupUpdates(
   }
   return res;
 }
+
+function addReplacementUpdateIfValid(
+  updates: LookupUpdate[],
+  config: LookupUpdateConfig
+): void {
+  const replacementNewName = determineNewReplacementName(config);
+  const replacementNewVValue = determineNewReplacementValue(config);
+
+  if (
+    config.packageName !== replacementNewName ||
+    config.currentValue !== replacementNewVValue
+  ) {
+    updates.push({
+      updateType: 'replacement',
+      newName: replacementNewName,
+      newValue: replacementNewVValue!,
+    });
+  }
+}
+
+function isReplacementNameRulesConfigured(config: LookupUpdateConfig): boolean {
+  if (
+    config.replacementName ||
+    config.replacementPrefixRemove ||
+    config.replacementPrefixAdd
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+function isReplacementRulesConfigured(config: LookupUpdateConfig): boolean {
+  if (isReplacementNameRulesConfigured(config) || config.replacementVersion) {
+    return true;
+  }
+
+  return false;
+}
+
+function determineNewReplacementName(config: LookupUpdateConfig): string {
+  let replacementNewName = config.packageName;
```

Comment:
:thinking_face: it should be possible to use early returns here. 

This would improve the readability.

Reply:
I did give this some thought when developing it, but thought it would actually make the conditional statements a lot more complex in order to have early returns.